canonicalwebteam/discourse_docs/parsers.py

```diff
@@ -85,21 +106,71 @@ def parse_topic(topic):
     }
 
 
-def parse_navigation(index_soup):
+def parse_navigation(index_soup, url_map):
     """
     Given the HTML soup of a index topic
-    extract the "navigation" section
+    extract the "navigation" section, and rewrite any
+    links in the url_map
     """
 
-    nav_soup = get_section(index_soup, "Content")
+    nav_soup = get_section(index_soup, "Navigation")
     nav_html = "Navigation missing"
 
     if nav_soup:
+        # Convert links to the form needed in this site
+        for link in nav_soup.find_all("a"):
+            if "href" in link.attrs:
+                url = link.attrs["href"]
+                link_match = TOPIC_URL_MATCH.match(url)
+
+                if link_match:
+                    topic_id = int(link_match.groupdict()["topic_id"])
+                    if topic_id in url_map:
+                        link.attrs["href"] = url_map[topic_id]
         nav_html = str(nav_soup)
 
     return nav_html
 
 
+def parse_url_map(index_soup):
+    """
+    Given the HTML soup of an index topic
+    extract the URL mappings from the "URLs" section
```

Comment:
Maybe link to the `HTML` that this builds upon?
I guess that any markup change could cause a failure?

Reply:
I'll provide an example snippet. I don't think linking to a specific implementation is the right thing to do.

Reply:
I've done this.